awswrangler/s3/_write_dataset.py

```diff
@@ -37,12 +51,33 @@ def _to_partitions(
         subdir = "/".join([f"{name}={val}" for name, val in zip(partition_cols, keys)])
         prefix: str = f"{path_root}{subdir}/"
         if mode == "overwrite_partitions":
-            delete_objects(
-                path=prefix,
-                use_threads=use_threads,
-                boto3_session=boto3_session,
-                s3_additional_kwargs=func_kwargs.get("s3_additional_kwargs"),
-            )
+            if (table_type == "GOVERNED") and (table is not None) and (database is not None):
+                del_objects: List[Dict[str, Any]] = _get_table_objects(
+                    catalog_id=catalog_id,
+                    database=database,
+                    table=table,
+                    transaction_id=transaction_id,  # type: ignore
+                    partition_cols=partition_cols,
+                    partitions_values=keys,
+                    partitions_types=partitions_types,
+                    boto3_session=boto3_session,
+                )
+                if del_objects:
+                    _update_table_objects(
+                        catalog_id=catalog_id,
+                        database=database,
+                        table=table,
+                        transaction_id=transaction_id,  # type: ignore
+                        del_objects=del_objects,
+                        boto3_session=boto3_session,
+                    )
+            else:
+                delete_objects(
+                    path=prefix,
+                    use_threads=use_threads,
+                    boto3_session=boto3_session,
+                    s3_additional_kwargs=func_kwargs.get("s3_additional_kwargs"),
+                )
```

Comment:
If `table_type` is "GOVERNED", mark files as deleted. Otherwise physically delete them. Same logic applies to partitions